### PR TITLE
bump FFI to 0.12.4

### PR DESCRIPTION
**Pull request by nbsp**
also exposing `Participant.disconnect_reason`

### CODE REVIEW

**Comment by theomonnom** [comment]
I'll try it with agents Tuesday. Let's also directly bump livekit-rtc inside this PR

**Review by davidzhao — APPROVED** [review]
this is working with examples and agents after a few fixes. merging it in

**Comment by davidzhao** [comment]
actually, types are broken.. would like to fix it before merging. @theomonnom any ideas?

https://github.com/livekit/python-sdks/actions/runs/12533404456/job/34953054932?pr=333

File: livekit-rtc/livekit/rtc/participant.py [hunk 1]
```diff
@@ -107,6 +107,11 @@ def kind(self) -> proto_participant.ParticipantKind.ValueType:
         """Participant's kind (e.g., regular participant, ingress, egress, sip, agent)."""
         return self._info.kind
 
+    @property
+    def disconnect_reason(self) -> proto_participant.DisconnectReason.ValueType:
```
Comment (theomonnom) [inline livekit-rtc/livekit/rtc/participant.py:111]: Maybe we should change this return `proto_participant.DisconnectReason.ValueType | None`

**Pull request merged**